Testing/SystemTests/tests/qt/SliceViewerIntegrationTest.py

```diff
@@ -0,0 +1,500 @@
+                                      OutputWorkspace='ws_MD_2d')
+    return histo_ws
+
+
+def create_histo_ws_positive():
+    histo_ws_positive = CreateMDHistoWorkspace(Dimensionality=3,
+                                               Extents='-3,3,-10,10,-1,1',
+                                               SignalInput=range(1, 101),
+                                               ErrorInput=range(100),
+                                               NumberOfBins='5,5,4',
+                                               Names='Dim1,Dim2,Dim3',
+                                               Units='MomentumTransfer,EnergyTransfer,Angstrom',
+                                               OutputWorkspace='ws_MD_2d_pos')
+    return histo_ws_positive
+
+
+def create_hkl_ws():
+    hkl_ws = CreateMDWorkspace(Dimensions=3,
+                               Extents='-10,10,-9,9,-8,8',
+                               Names='A,B,C',
+                               Units='r.l.u.,r.l.u.,r.l.u.',
+                               Frames='HKL,HKL,HKL',
+                               OutputWorkspace='hkl_ws')
+    expt_info = CreateSampleWorkspace()
+    hkl_ws.addExperimentInfo(expt_info)
+    SetUB(hkl_ws, 1, 1, 1, 90, 90, 90)
+    return hkl_ws
+
+
+class HelperTestingClass(QtWidgetFinder):
+    """
+    Base class for system tests for testing sliceviewer integration with qt and matplotlib.
+    """
+    def __init__(self):
+        self._qapp = get_application()
+
+    def cleanup(self):
+        for ii in self._qapp.topLevelWidgets():
+            ii.close()
+        self._qapp.sendPostedEvents()
+        self._qapp.sendPostedEvents()
+        self.assert_no_toplevel_widgets()
+        self._qapp = None
+
+    # private methods
+    def _assertNoErrorInADSHandlerFromSeparateThread(self, operation):
+        """Check the error stream for any errors when calling operation.
+        Raise an assertion error if errors are detected
+        """
+        try:
+            # the ads handler catches all exceptions so that the handlers don't
+            # bring down the sliceviewer. Check if anything is written to stderr
+            stderr_capture = io.StringIO()
+            stderr_orig = sys.stderr
+            sys.stderr = stderr_capture
+            op_thread = Thread(target=operation)
+            op_thread.start()
+            op_thread.join()
+            self.assertTrue('Error occurred in handler' not in stderr_capture.getvalue(),
+                            msg=stderr_capture.getvalue())
+        finally:
+            sys.stderr = stderr_orig
+
+
+class SliceViewerTestDeleteOnClose(systemtesting.MantidSystemTest, HelperTestingClass):
+    def runTest(self):
+        HelperTestingClass.__init__(self)
+        histo_ws = create_histo_ws()
+        pres = SliceViewer(histo_ws)
+        self.assert_widget_created()
+        pres.view.close()
+
+        self._qapp.sendPostedEvents()
+        self._qapp.sendPostedEvents()
```

Comment:
Does this happen twice for a reason?

Reply:
I actually have no idea but that's how it was in the original tests. @martyngigg Do you know if this was deliberate?

Reply:
You're not going to like the answer...
`sendPostedEvents` didn't always appear to process all of the events with a single call. I expect some interaction with the Python GIl might been some Qt events are not sent before the sendPostedEvents call requiring a second one. This seemed to get worse in some cases with Qt5 too.

Reply:
Ok, I guess we'll just leave it for now then